servers/services/src/main/java/org/projectnessie/services/rest/BaseResource.java

```diff
@@ -60,63 +58,44 @@ protected BaseResource(
   }
 
-  Optional<WithHash<Ref>> getRefWithHash(String ref) {
-    try {
-      WithHash<Ref> whr = store.toRef(Optional.ofNullable(ref).orElse(config.getDefaultBranch()));
-      return Optional.of(whr);
-    } catch (ReferenceNotFoundException e) {
+  Optional<NamedRef> optionalNamedRefOrThrow(String ref) throws NessieNotFoundException {
+    if (ref == null || ref.isEmpty()) {
       return Optional.empty();
     }
+    return Optional.of(namedRefWithHashOrThrow(ref).getValue());
   }
 
-  WithHash<NamedRef> namedRefWithHashOrThrow(String namedRef, @Nullable String hashOnRef)
-      throws NessieNotFoundException {
-    List<WithHash<NamedRef>> collect =
-        store
-            .getNamedRefs()
-            .filter(
-                r ->
-                    r.getValue().getName().equals(namedRef)
-                        || r.getValue().getName().equals(config.getDefaultBranch()))
-            .collect(Collectors.toList());
-    WithHash<NamedRef> namedRefWithHash;
-    if (collect.size() == 1) {
-      namedRefWithHash = collect.get(0);
-    } else {
-      namedRefWithHash =
-          collect.stream()
-              .filter(r -> r.getValue().getName().equals(namedRef))
-              .findFirst()
-              .orElseThrow(
-                  () ->
-                      new NessieNotFoundException(String.format("Ref for %s not found", namedRef)));
-    }
-
+  WithHash<NamedRef> namedRefWithHashOrThrow(String ref) throws NessieNotFoundException {
     try {
-      if (null == hashOnRef) {
-        return namedRefWithHash;
+      if (null != ref) {
+        ref = Validation.validateReferenceName(ref);
       }
-
-      // we need to make sure that the hash in fact exists on the named ref
-      Hash hash = Hash.of(hashOnRef);
-      if (store.getCommits(namedRefWithHash.getValue()).noneMatch(c -> c.getHash().equals(hash))) {
-        throw new NessieNotFoundException(
-            String.format("Hash %s on Ref %s could not be found", hashOnRef, namedRef));
-      }
-      return WithHash.of(hash, namedRefWithHash.getValue());
+      return store.toRef(Optional.ofNullable(ref).orElse(config.getDefaultBranch()));
     } catch (ReferenceNotFoundException e) {
-      throw new NessieNotFoundException(
-          String.format("Hash %s on Ref %s could not be found", hashOnRef, namedRef));
+      throw new NessieNotFoundException(String.format("Ref for %s not found", ref));
     }
   }
 
-  Hash getHashOrThrow(String ref) throws NessieNotFoundException {
-    return getHash(ref)
-        .orElseThrow(() -> new NessieNotFoundException(String.format("Ref for %s not found", ref)));
+  static Optional<Hash> toHash(String hash, boolean required) throws NessieConflictException {
+    try {
+      return toHashIA(hash, required);
+    } catch (IllegalArgumentException e) {
+      throw new NessieConflictException("Must provide expected hash value for operation.");
+    }
+  }
+
+  static Optional<Hash> toHashIA(String hash, boolean required) {
```

Comment:
can you expland the `IA` please? Additionally, the throws confuse me here. What am I missing